File: server/src/agents/exampleAgent.js
import BaseAgent from './baseAgent.js';

export default class ExampleAgent extends BaseAgent {
  constructor(opts = {}) {
    super({ name: 'example', ...opts });
  }

  async run(input) {
    // Very small example: echo with metadata
    return {
      agent: this.name,
      input,
      output: `Echo: ${String(input).slice(0, 1024)}`,
      ts: new Date().toISOString(),
    };
  }
}

File: ml_service/agents/example_agent.py
class ExampleAgent:
    def __init__(self, name: str = 'py-example'):
        self.name = name

    def run(self, prompt: str):
        # simple sync implementation for now
        return {
            'agent': self.name,
            'input': prompt,
            'output': f'ECHO_PY: {prompt[:1024]}',
            'len': len(prompt),
        }
